Source/WebCore/bindings/js/JSEventListener.h

```diff
@@ -79,13 +81,14 @@ class JSEventListener : public EventListener, public JSVMClientData::Client {
 protected:
     enum class CreatedFromMarkup : bool { No, Yes };
 
-    JSEventListener(JSC::JSObject* function, JSC::JSObject* wrapper, bool isAttribute, CreatedFromMarkup, DOMWrapperWorld&);
+    JSEventListener(JSC::JSObject* function, JSC::JSObject* wrapper, bool isAttribute, CreatedFromMarkup, DOMWrapperWorld&, bool isWindowOnError);
     void handleEvent(ScriptExecutionContext&, Event&) override;
     void setWrapperWhenInitializingJSFunction(JSC::VM&, JSC::JSObject* wrapper) const { m_wrapper = JSC::Weak<JSC::JSObject>(wrapper); }
 
 private:
     bool m_isAttribute : 1;
     bool m_wasCreatedFromMarkup : 1;
+    const bool m_isWindowOnError;
```

Comment:
I'm asking this question on Slack but let me repost it here and discuss here instead.

The root cause for the issue is that WebKit is always using `JSLazyEventListener` in `attributeChanged` methods (e.g. in [`HTMLBodyElement`](https://github.com/WebKit/WebKit/blob/bfad000ee7c8b2b89733b2f243d1db130940ad8e/Source/WebCore/html/HTMLBodyElement.cpp#L167))

https://github.com/WebKit/WebKit/blob/bfad000ee7c8b2b89733b2f243d1db130940ad8e/Source/WebCore/dom/Document.cpp#L5405-L5412

While I currently can fix the issue by adding a boolean flag in `JSEventListener`, I wonder I should use `JSErrorHandler` in `Document::setWindowAttributeEventListener` instead. However, it seems a proper `JSObject listener` is required to initialize it but I couldn't find a way to initialize such object. Do you have any suggestions?

Or, should I try to create `JSLazyErrorHandler` and use it in `Document::setWindowAttributeEventListener` instead?